### ISSUE TITLE
max test accuracy: 0

### ISSUE THREAD

**Issue by NatLun091238** [martin-gorner/tensorflow-mnist-tutorial#40]
Hi, thank you for very nice tutorials!! 

I get the following when run the code:
runfile('C:/Users/natlun/Documents/Python Scripts/Recognition/tensorflow-mnist-tutorial-master/mnist_4.2_batchnorm_convolutional.py', wdir='C:/Users/natlun/Documents/Python Scripts/Recognition/tensorflow-mnist-tutorial-master')
Reloaded modules: tensorflowvisu_digits, tensorflowvisu
Extracting data\train-images-idx3-ubyte.gz
Extracting data\train-labels-idx1-ubyte.gz
Extracting data\t10k-images-idx3-ubyte.gz
Extracting data\t10k-labels-idx1-ubyte.gz

￼
max test accuracy: 0

How to fix that issue??

**Comment by NatLun091238** [comment]
pip install --upgrade matplotlib
Requirement already up-to-date: matplotlib in c:\users\natlun\appdata\local\continuum\anaconda3\lib\site-packages
Requirement already up-to-date: pyparsing!=2.0.4,!=2.1.2,!=2.1.6,>=1.5.6 in c:\users\natlun\appdata\local\continuum\anaconda3\lib\site-packages (from matplotlib)
Requirement already up-to-date: python-dateutil in c:\users\natlun\appdata\local\continuum\anaconda3\lib\site-packages (from matplotlib)
Requirement already up-to-date: six>=1.10 in c:\users\natlun\appdata\local\continuum\anaconda3\lib\site-packages (from matplotlib)
Requirement already up-to-date: numpy>=1.7.1 in c:\users\natlun\appdata\local\continuum\anaconda3\lib\site-packages (from matplotlib)
Requirement already up-to-date: pytz in c:\users\natlun\appdata\local\continuum\anaconda3\lib\site-packages (from matplotlib)
Requirement already up-to-date: cycler>=0.10 in c:\users\natlun\appdata\local\continuum\anaconda3\lib\site-packages (from matplotlib)

**Comment by martin-gorner** [comment]
I have seen this error on Linux laptops that had matplotlib installed but did not have a graphic backend that matplotlib could use to display somehting.

On linux, you have to install matplotlib with apt-get:
sudo -H apt-get install python3-matplotlib

and then sometimes the version of matplotlib in the apt package is stale so I still recommend to update it:
sudo -H pip3 install --upgrade matplotlib

**Comment by NatLun091238** [comment]
Thank you for taking your time to respond. Unfortunately, I run Python and TensorFlow on Windows 10. So far I didn't figure out which dependencies of matplotlib are missing.

**Comment by NatLun091238** [comment]
The problem was in Spyder settings (solution: http://stackoverflow.com/questions/23585126/how-do-i-get-interactive-plots-again-in-spyder-ipython-matplotlib). The issue is solved by now!

**Comment by martin-gorner** [comment]
I'm glad you were able to resolve this.

**Comment by Sankar321** [comment]
import tensorflow as tf 


import os
os.environ['TF_CPP_MIN_LOG_LEVEL'] = '2'



from tensorflow.examples.tutorials.mnist import input_data

mnist = input_data.read_data_sets("/tmp/data",one_hot=True)


x = tf.placeholder(tf.float32,[None,28,28,1])
w = tf.Variable(tf.zeros([784, 10]))
b = tf.Variable(tf.zeros([10]))

init = tf.initialize_all_variables()

#model

x = tf.reshape(x,[-1,784])

y = tf.nn.softmax(tf.matmul(x,w) + b)

y_ = tf.placeholder(tf.float32,[None,10])

cross_entropy = -tf.reduce_sum(y_* tf.log(y))

is_correct = tf.equal(tf.argmax(y,1),tf.argmax(y_,1))

accuracy = tf.reduce_mean(tf.cast(is_correct,tf.float32))


#Training

optimizer = tf. train.GradientDescentOptimizer(0.003)

trainstep = optimizer.minimize(cross_entropy)


sess = tf.Session()

sess.run(init)


for i in range(100):

	batch_x, batch_y = mnist.train.next_batch(100)
	train_data = {x: batch_x, y_:batch_y}

	sess.run(trainstep, feed_dict = train_data)

	a,c = sess.run([accuracy,cross_entropy], feed_dict = train_data)

	print('Accuracy:',a,'cross_entropy:',c)
	
	





for i in range(100):

	test_data = {x: mnist.test.images, y_: mnist.test.labels}
	a,c = sess.run([accuracy, cross_entropy,], feed_dict=test_data)

	print('Test_Accuracy:',a,'Test_cross_entropy:',c)


sess.close()

Respected Martin sir,


In the above code, I am getting very high values( starts at 4000) for cross entropy on test data and getting reduced to 2000

How this high values for entropy on testing images?

Please help me sir
I am new to this area
